internal/services/apimanagement/api_management_policy_fragment_resource_test.go

```diff
@@ -0,0 +1,182 @@
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type ApiManagementPolicyFragmentResource struct{}
+
+func TestAccApiManagementPolicyFragment_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_management_policy_fragment", "test")
+	r := ApiManagementPolicyFragmentResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("description").HasValue("Some descriptive text"),
+				check.That(data.ResourceName).Key("format").HasValue("xml"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccApiManagementPolicyFragment_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_management_policy_fragment", "test")
+	r := ApiManagementPolicyFragmentResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func TestAccApiManagementPolicyFragment_updateDescription(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_management_policy_fragment", "test")
+	r := ApiManagementPolicyFragmentResource{}
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("description").HasValue("Some descriptive text"),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.update(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("description").HasValue("Some descriptive text which is updated"),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccApiManagementPolicyFragment_rawxml(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_api_management_policy_fragment", "test")
+	r := ApiManagementPolicyFragmentResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.rawxml(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func (ApiManagementPolicyFragmentResource) Exists(ctx context.Context, clients *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := policyfragment.ParsePolicyFragmentIDInsensitively(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := clients.ApiManagement.PolicyFragmentClient.Get(ctx, *id, policyfragment.GetOperationOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("retrieving %s: %+v", id, err)
+	}
+
+	return pointer.To(resp.Model != nil && resp.Model.Id != nil), nil
```

Comment:
in this case presuming we get an object back is sufficient to confirm this exists

```suggestion
	return pointer.To(resp.Model != nil), nil
```